セッション周り(サインアップ、サインアウト、)の実装

yosuketokunaga opened the pull request:
#  WHAT
サインアップ、サインイン、サインアウト機能の実装 

#  WHY
ユーザー登録に必要なため。

Commit:
create pull request

Commit:
add_culums_to_userstable

・add columns to users table
・add @import user.scss to application.scss
・change erb files to haml

yosuketokunaga commented:
![2017-04-08 11 09 25](https://cloud.githubusercontent.com/assets/26739922/24824771/0f59066c-1c4d-11e7-85fd-5fbeec467e5e.png)

yosuketokunaga commented:
![2017-04-08 11 09 33](https://cloud.githubusercontent.com/assets/26739922/24824776/1a93c774-1c4d-11e7-9e97-c616aad3c4f4.png)

sato11 commented on app/controllers/application_controller.rb:
```diff
@@ -2,4 +2,9 @@ class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
+  before_action :configure_permitted_parameters, if: :devise_controller?
+
+  def configure_permitted_parameters
```
このメソッドは private 等にした方がいいのではないでしょうか？

yosuketokunaga replied:
追記しました。

sato11 commented on app/assets/stylesheets/user.scss:
```diff
@@ -0,0 +1,141 @@
+        display: block;
+        width: 60px;
+        margin-top: 5px;
+        border-bottom: 1px solid #38aef0;
```
カラーコードについてですが、直接記述すると後々になって管理コストが発生することが予想されるため、色変数を利用して管理してみましょう。

yosuketokunaga replied:
こちら変数を設定し、呼び出すよう修正致しました。

sato11 commented on app/assets/stylesheets/user.scss:
```diff
@@ -0,0 +1,141 @@
+    float: left;
+    margin-top: 20px;
+    color: #fff;
+    background-color: #38aef0;
```
例えば 33行目のここは同じカラーコードですが、双方を変更したい時にこのような記述になっておりますと万が一変更するとなった際にそれだけで倍の手間がかかってしまいます。
またこのカラーコードが具体的にどのような色なのかがこれだけでは分からなくなってしまいます。

yosuketokunaga replied:
こちらも変数を利用するよう修正致しました。

sato11 commented on config/routes.rb:
```diff
@@ -1,4 +1,5 @@
   devise_for :users
-    get 'messages' => 'messages#index'
+    root  'messages#index'
+    get   'messages' => 'messages#index'
```
ルーティングを生成する際に非常に有用なメソッドがあったかと思いますが、お覚えでしょうか？
ぜひそちらを調べて、利用してみていただきたいです！

yosuketokunaga replied:
get '~'の記述をresourcesを用いて記述し直しました。

sato11 reviewed (commented):
画面遷移後にフラッシュメッセージの表示はございますでしょうか？
ありましたらそちらの画像も共有いただけると嬉しいです！

yosuketokunaga commented:
このpullrequestではセッション周りの実装のみ行い、フラッシュメッセージの作成はまた別のpullrequest出させていただきます。ご確認お願いいたします!

Commit:
change routes.rb

・change describe routes.rb
・add private method to application_controller
・use variable to color

sato11 commented on app/controllers/application_controller.rb:
```diff
@@ -2,4 +2,10 @@ class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
+  before_action :configure_permitted_parameters, if: :devise_controller?
+
+private
```
private のインデントは一般的に同じ高さに揃えるのがベターです。

yosuketokunaga replied:
privateのインデント位置修正致しました。

sato11 commented on app/controllers/application_controller.rb:
```diff
@@ -2,4 +2,10 @@ class ApplicationController < ActionController::Base
+
+private
+  def configure_permitted_parameters
+      devise_parameter_sanitizer.permit(:sign_up, keys: [:name])
```
「アカウントの編集」においてnameの変更が反映されないのではないでしょうか？

yosuketokunaga replied:
記述を変更いたしました。ご確認お願いいたします。

sato11 commented on config/routes.rb:
```diff
@@ -1,5 +1,5 @@
-    root  'messages#index'
-    get   'messages' => 'messages#index'
+  root  'messages#index'
+  resources :messages
```
これだと7つのリソース全てが生成されてしまうので、必要なアクションの分だけ指定するのがいいですね。

yosuketokunaga replied:
現時点で必要であるindexのみ生成する記述に変更いたしました。

Commit:
Edit.aplication_controllers

add_name for sign_up & account_update

sato11 commented on app/controllers/application_controller.rb:
```diff
@@ -2,4 +2,11 @@ class ApplicationController < ActionController::Base
+  private
+  def configure_permitted_parameters
+      devise_parameter_sanitizer.for(:sign_up) << :name
+      devise_parameter_sanitizer.for(:account_update) << :name
```
非常に細かいのですが、今度はこちらのインデントがやや深くなっていますね。
ruby のコーディングスタンダードは半角スペース二つ分のネストに統一することとされています！

sato11 replied:
ちなみにこちらを変更された理由はなんでしょうか？

for を使った書き方は旧いバージョンの devise における書き方かと思われます。
個人的にはpermitを使った書き方を推奨させていただきたく思います。

変更に当たって参考にしたサイト等ありましたらお教え願えますか？

yosuketokunaga replied:
インデントについて修正致しました。

変更についてですが、以下のサイトを参考にしました。
https://teratail.com/questions/39837
更新日時も一年以内のものだったので正しい情報と思い変更しておりました。
.permitでの記述に再度修正致しました。

Commit:
change strongparameter

・fix indent
・fix　description
